modules/standard/Random.chpl

```diff
@@ -250,44 +250,47 @@ module Random {
         compilerError('choice() prob array must be 1 dimensional');
       }
 
-      if prob.domain != arr.domain {
-        throw new owned IllegalArgumentError('choice() array arguments must have same domain');
+      //Commenting out for now, as X is domain of reindexed array.
```

Comment:
How should we handle this, as the original array is reindexed in the overload function itself?

Reply:
see other answer